UnityProject/Assets/Scripts/Systems/Antagonists/Antags/Changeling/ChangelingMain.cs

```diff
@@ -0,0 +1,643 @@
+		[SyncVar] private float chemMax = 75;
+		[SyncVar] private float chemAddPerTime = CHEM_ADD_TIME;
+		[SyncVar] private float chemAddTime = CHEM_ADD_PER_TIME_BASE;
+		public int ExtractedDNA => changelingDNAs.Count;
+
+		[SyncVar(hook = nameof(SyncResetCounts))] private int resetCount = 0;
+		[SyncVar(hook = nameof(SyncMaxResetCounts))] private int resetCountMax = 1;
+
+		[SyncVar] private int absorbCount = 0;
+		public int AbsorbCount => absorbCount;
+		public int ResetsLeft => resetCountMax - resetCount;
+
+		private List<ChangelingDNA> changelingDNAs = new List<ChangelingDNA>();
+		private List<ChangelingMemories> changelingMemories = new List<ChangelingMemories>();
+		public List<ChangelingMemories> ChangelingMemories => changelingMemories;
+
+		public bool isFakingDeath = false;
+
+		public ChangelingDNA currentDNA;
+
+		[SyncVar(hook = nameof(ChangelingDNASync))]
+		private string changelingDNASer = "";
+		[SyncVar(hook = nameof(ChangelingMemoriesSync))]
+		private string changelingMemoriesSer = "";
+
+		public List<ChangelingDNA> ChangelingDNAs => new List<ChangelingDNA>(changelingDNAs);
+		public List<ChangelingDNA> ChangelingLastDNAs
+		{
+			get
+			{
+				return ChangelingDNAs.Skip(changelingDNAs.Count - MAX_LAST_EXTRACTED_DNA_FOR_TRANSFORM).ToList();
+			}
+		}
+		[SyncVar(hook = nameof(SyncEPCount))] private int evolutionPoints = 10;
+		public int EvolutionPoints => evolutionPoints;
+
+		private ObservableCollection<ChangelingAbility> abilitiesNow => changelingMind.ChangelingAbilities;
+		public ObservableCollection<ChangelingAbility> AbilitiesNow => abilitiesNow;
+		private List<ChangelingData> AbilitiesNowDataSynced = new List<ChangelingData>();
+		public List<ChangelingData> AbilitiesNowData
+		{
+			get
+			{
+				return AbilitiesNowDataSynced;
+			}
+		}
+		public List<ChangelingData> AllAbilities => ChangelingAbilityList.Instance.Abilites;
+
+		[SyncVar(hook = nameof(SyncAbilityList))]
+		private string abilitesIDSNow = "";
+
+		public static Dictionary<uint, ChangelingMain> ChangelingByMindID = new();
+		public static Dictionary<uint, Mind> ChangelingMinds = new();
+		private UI_Changeling uiChangeling;
+		public UI_Changeling Ui => uiChangeling;
+		private Mind changelingMind;
+		public Mind ChangelingMind
+		{
+			get
+			{
+				if (changelingMind == null)
+				{
+					Logger.LogError("Changeling can`t find Mind", Category.Changeling);
+					changelingMind = gameObject.transform.parent.gameObject.GetComponent<PlayerScript>().Mind;
+				}
+				return changelingMind;
+			}
+		}
+
+		public List<ActionData> ActionData => throw new NotImplementedException();
+
+		PlayerScript playerScript;
+
+		private int tickTimer = 0;
+
+		private const float CHEM_ADD_TIME = 1;
+		private const float CHEM_ADD_PER_TIME_BASE = 2;
+		private const float CHEM_ADD_PER_TIME_BASE_SLOWED = 1;
+		private const int MAX_LAST_EXTRACTED_DNA_FOR_TRANSFORM = 7;
+
+		#region Hooks
+
+		[Client]
```

Comment:
why are you putting `[Client]` on syncvar hooks?

Reply:
Just copied that from other scripts